### PR TITLE
Apply material to BRep based direct shapes

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

Apply material to BRep based direct shapes. We somehow missed this when we added support for BRep based direct shapes. This will fix MAGN-10141 Brep based direct shape doesn't display with material (#1004 Direct Shape Materials not Displaying in Revit 2017).

This fix is for Revit 2017 only.

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.

### Reviewers
@QilongTang 

### FYIs
@mjkkirschner @jnealb

### CODE REVIEW

File: src/Libraries/RevitNodes/Elements/DirectShape.cs [hunk 1]
```diff
@@ -221,17 +213,17 @@ private static IList<GeometryObject> GenerateTessellatedGeo(DesignScriptEntity s
             else if (shapeReference is Autodesk.DesignScript.Geometry.Surface)
             {
                 // Defer to tessellated shape if BRep shape for some reason fails
-                GeometryObject geometry = null; 
+                GeometryObject geometry = null;
                 try
                 {
-                    geometry = DynamoToRevitBRep.ToRevitType(shapeReference as Autodesk.DesignScript.Geometry.Surface);
+                    geometry = DynamoToRevitBRep.ToRevitType(shapeReference as Autodesk.DesignScript.Geometry.Surface, true, materialId);
```
Comment (sm6srw) [inline src/Libraries/RevitNodes/Elements/DirectShape.cs]: Pass material Id.

File: src/Libraries/RevitNodes/Elements/DirectShape.cs [hunk 2]
```diff
@@ -243,17 +235,17 @@ private static IList<GeometryObject> GenerateTessellatedGeo(DesignScriptEntity s
             else if (shapeReference is Autodesk.DesignScript.Geometry.Solid)
             {
                 // Defer to tessellated shape if BRep shape for some reason fails
-                GeometryObject geometry = null; 
+                GeometryObject geometry = null;
                 try
                 {
-                    geometry = DynamoToRevitBRep.ToRevitType(shapeReference as Autodesk.DesignScript.Geometry.Solid);
+                    geometry = DynamoToRevitBRep.ToRevitType(shapeReference as Autodesk.DesignScript.Geometry.Solid, true, materialId);
```
Comment (sm6srw) [inline src/Libraries/RevitNodes/Elements/DirectShape.cs]: Pass material Id.

File: src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs [hunk 3]
```diff
@@ -1,138 +1,146 @@
...
+        /// <summary>
+        /// this method attempts to construct a BRep from a surface.
+        /// </summary>
+        /// <param name="surf"></param>
+        /// <param name="performHostUnitConversion"></param>
+        /// <param name="materialId"></param>
+        /// <returns></returns>
+        public static GeometryObject ToRevitType(Surface surf,
+          bool performHostUnitConversion = true,
+          ElementId materialId = null)
```
Comment (sm6srw) [inline src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs]: Add material id param as an optional param the end in order to maintain backward compability.

File: src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs [hunk 4]
```diff
@@ -1,138 +1,146 @@
...
+        /// <summary>
+        /// this method attempts to construct a BRep from a closed solid.
+        /// </summary>
+        /// <param name="sol"></param>
+        /// <param name="performHostUnitConversion"></param>
+        /// <param name="materialId"></param>
+        /// <returns></returns>
+        public static GeometryObject ToRevitType(Solid sol,
+          bool performHostUnitConversion = true,
+          ElementId materialId = null)
```
Comment (sm6srw) [inline src/Libraries/RevitNodes/GeometryConversion/DynamoToRevitBRep.cs]: Add material id param as an optional param the end in order to maintain backward compability.

File: test/Libraries/RevitNodesTests/Elements/DirectShapeTests.cs [hunk 5]
```diff
@@ -96,6 +96,8 @@ private void ByBrepNameCategoryMaterial_Cuboid()
             Assert.AreEqual(10, geo.Faces.Size);
             // Check number of Edges
             Assert.AreEqual(24, geo.Edges.Size);
+            // Check that material is set
+            Assert.AreNotEqual(-1, geo.Faces.get_Item(0).MaterialElementId.IntegerValue);
```
Comment (sm6srw) [inline test/Libraries/RevitNodesTests/Elements/DirectShapeTests.cs]: Make sure Material Id is set on BRep based direct shapes.

**Comment by QilongTang** [comment]
LGTM. Thanks for the code format and usings clean up as well.

**Pull request merged**